mslib/msui/kmloverlay_dockwidget.py

```diff
@@ -319,59 +321,72 @@ def __init__(self, parent=None, view=None):
                 del self.dict_files[file]  # remove non-existent files from dictionary
             self.load_file()
 
-        palette = QtGui.QPalette(self.dialog.pushButton_colour.palette())
+        palette = QtGui.QPalette(self.pushButton_color.palette())
         colour = QtGui.QColor()
         colour.setRgbF(*settings["colour"])
         palette.setColor(QtGui.QPalette.Button, colour)
-        self.dialog.pushButton_colour.setPalette(palette)  # sets the last colour before closing KML Overlay
-        self.dialog.dsb_linewidth.valueChanged.connect(self.select_linewidth)
-
-    def open_customize_kml_dialog(self):
-        file = self.listWidget.currentItem().text()
-        # Set the colour of the Colour button to the colour of specific KML plot
-        if self.dict_files[file]["color"] is not None:
-            palette = QtGui.QPalette(self.dialog.pushButton_colour.palette())
-            colour = QtGui.QColor()
-            colour.setRgbF(*self.set_color(file))
-            palette.setColor(QtGui.QPalette.Button, colour)
-            self.dialog.pushButton_colour.setPalette(palette)
-        # Set the linewidth value to the linewidth of specific KML plot
-        if self.dict_files[file]["linewidth"] is not None:
-            self.dialog.dsb_linewidth.setValue(self.set_linewidth(file))
-
-        self.dialog.show()
+        self.pushButton_color.setPalette(palette)  # sets the last colour before closing KML Overlay
+
+        # When KMLoverlaywidget is opened,it ensures that the color of individual KML files are already shown as icons.
+        self.set_color_icons()
+        self.flag = 0  # set flag to know prevent same icon be set twice when we want to show linewidth of KML files
+        # (flag prevents valueChanged connect to function)
 
     def __del__(self):  # destructor
         for x in self.dict_files:
             self.dict_files[x]["patch"] = None  # patch object has to be deleted
         settings = {
             "filename": str(self.directory_location),
-            "linewidth": self.dialog.dsb_linewidth.value(),
+            "linewidth": self.dsbx_linewidth.value(),
             "colour": self.get_color(),
             "saved_files": self.dict_files
         }
         save_settings_qsettings(self.settings_tag, settings)
 
+    def set_color_icons(self):
+        """
+        Shows the color of individual KML files set as icons before them whose heights
+        are in correspondence to the linewidth(thickness) of the KML file.
+        """
+
+        if(self.listWidget.count() != 0):
```

Comment:
This looks like C. Please follow the PEP8 style guidelines (https://www.python.org/dev/peps/pep-0008/). You can use the flake8 tool to test your code for style and syntax errors. The merge request cannot be accepted until all tests pass, one of which is the flake8 tool.

Reply:
flake8 was successful. but the style ( ) is not needed here

Reply:
Sorry for this style ( ) . I will remove those.

Reply:
The style ( ) for if has been removed.